services/toDoService.js

```diff
@@ -14,3 +32,59 @@ export async function getAllToDos() {
 
   return sanitizedToDos;
 }
+
+export async function getToDosByDestinationId(destinationId) {
+  if (!validateId(destinationId)) return { status: 400, error: "Invalid destinationId" };
+  await dbConnect();
+
+  const toDos = await ToDo.find({ destinationId: destinationId });
+  if (!Array.isArray(toDos)) return { status: 500, error: "Internal server error" };
+  const sanitizedToDos = toDos.map((toDo) => ({
+    id: toDo.id,
+    description: toDo.description,
+    checked: toDo.checked,
+    destinationId: toDo.destinationId,
+  }));
+
+  return sanitizedToDos;
+}
+
+export async function createToDo(body, destinationId) {
+  if (!validateId(destinationId)) return { status: 400, error: "Invalid destinationId" };
+  await dbConnect();
+
+  const newToDo = await ToDo.create({
+    description: body.description,
+    destinationId: destinationId,
+  });
+
+  return await getToDosByDestinationId(destinationId);
+}
+
+export async function updateToDo(id, body) {
+  if (!validateId(id)) return { status: 400, error: "Invalid id" };
+
+  await dbConnect();
+  const toDo = await ToDo.findById(id);
+  if (!toDo) return { status: 404, error: id + " not found" };
+  if (body.description) toDo.description = body.description;
+  toDo.checked = body.checked;
+  const savedToDo = await toDo.save();
+  if (!savedToDo) return { status: 500, error: "Internal server error" };
+  return await getToDosByDestinationId(toDo.destinationId);
+}
+
+export async function deleteToDo(id, destinationId) {
+  if (!validateId(id)) return { status: 400, error: "Invalid id" };
```

Comment:
This should probably not be checked here. You should check for ids when creating a new Todo, or let them be created by mongoose and not check for them to be valid at all. Your server should not crash here if this method is being called with an invalid id.

Reply:
Oops, that makes sense. I removed the validation checks for non-creating operations and change it to prove if the object with that id exists in db.